src/mono/wasm/runtime/library_mono.js

```diff
@@ -779,6 +779,49 @@ var MonoSupportLib = {
 			// and nested class names like Foo/Bar to Foo.Bar
 			return className.replace(/\//g, '.').replace(/`\d+/g, '');
 		},
+
+		mono_wasm_load_data: function (data, prefix, magic) {
+			var dataview = new DataView(data.buffer);
+			var sig = dataview.getUint32(0, true);
+			//	get magic number
+			if (sig != magic) {
+				throw new Error ("File is of wrong type");
+			}
+			var manifestSize = dataview.getUint32(4, true);
+			var manifestContent = Module.UTF8ArrayToString(data, 8, manifestSize);
+			var manifest = JSON.parse(manifestContent);
+			data = data.slice(manifestSize+8);
+
+			// Create the folder structure
+			// /usr/share/zoneinfo
+			// /usr/share/zoneinfo/Africa
+			// /usr/share/zoneinfo/Asia
+			// ..
+			var p = prefix.slice(1).split('/');
+			p.forEach((v, i) => {
+				FS.mkdir(v);
+				Module['FS_createPath']("/" + p.slice(0, i).join('/'), v, true, true);
+			})
+			var folders = new Set()
+			manifest.filter(m => {
+				m = m[0].split('/')
+				if (m!= null) {
+					if (m.length === 3) folders.add(m.slice(0,2).join('/'));
```

Comment:
This appears to limit the depth of the directory structure in the archive, can you write it so that it works with arbitrary depths?